sdk/core/azure-core-amqp/inc/azure/core/amqp/connection_string_credential.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-Licence-Identifier: MIT
+
+#pragma once
+
+#include "network/sasl_transport.hpp"
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Core { namespace _internal { namespace Amqp {
+  //
+  // A ServiceBus connection string has the following format:
+  // "Endpoint=sb://<namespace>.servicebus.windows.net/;SharedAccessKeyName=<KeyName>;SharedAccessKey=<KeyValue>;EntityPath=<entity>"
+  //
+
+  enum class CredentialType
+  {
+    None,
+    SaslPlain,
+    ServiceBusSas,
+    BearerToken,
+  };
+
+  class ConnectionStringCredential {
```

Comment:
delete copy constructor and assignment operator

Reply:
Why? Is there a reason these can't be copied?

Reply:
It is inheritable (and is designed to be a base class), so slicing will be a problem. If you want derived classes to be able to have copy constructors and assignment operators, move them to protected.